### PR TITLE
Fixing basic Ubuntu CMake Build

### PR DESCRIPTION
**Pull request by rbharath**
PR #728 added basic CMake support for Tensorflow. However, that PR was only tested on MacOSX, so had a few errors on Ubuntu. This PR adds in fixes to make basic model compile on Ubuntu 12.04, with gcc 4.9.2, glibc 2.15.

The compile model does not yet run to completion (segfaults). I hope to debug that issue in a follow-on PR.

### CODE REVIEW

**Comment by rbharath** [comment]
I signed it!

**Comment by martinwicke** [comment]
Jenkins, test this please.

Testing for core/lib/io change.

File: tensorflow/core/lib/io/record_reader.h [hunk 1]
```diff
@@ -21,6 +21,16 @@ limitations under the License.
 #include "tensorflow/core/platform/macros.h"
 #include "tensorflow/core/platform/types.h"
 
+// SIZE_MAX is not defined for older gcc versions (5.1 and back). Define if
+// necessary.
+#ifndef SIZE_MAX
```
Comment (vrv) [inline tensorflow/core/lib/io/record_reader.h]: I wonder if this should be somewhere else, like in macros or types.h though

**Comment by martinwicke** [comment]
Our tests are down for the weekend, so we'll have to wait a bit. Sorry about that.

File: tensorflow/core/lib/io/record_reader.h (hunk 1 above)
Comment (rbharath) [inline tensorflow/core/lib/io/record_reader.h]: Good suggestion. Moved this to `macros.h`

**Comment by rbharath** [comment]
@martinwicke No problem. I made a couple small fixes in the meanwhile. As a quick addendum, what would be the best way to get the cmake build onto the jenkins test? I found that as I was making changes, I accidentally broke the build a couple times. It would be convenient to have jenkins catch this.

**Comment by jendap** [comment]
@rbharath can you please squash the commits? It would be great to have cmake working! I will try to set up jenkins build for cmake later this week.

**Comment by rbharath** [comment]
@jendap Just squashed the commits. Great to hear you have time to get the cmake build up on jenkins! Please let me know if I can help. Right now the cmake build is pretty rudimentary. The tutorial compiles on Mac OSX/Ubuntu, but segfaults for some reason that needs to be fixed. We also need to add SWIG and GPU support to the cmake build. I'll try to work on this some over the next couple weeks, but would be great if any other folks are interested in helping :)

**Comment by rbharath** [comment]
Oops, looks like my squashing messed things up here. I'll fix this and update this thread once everything is sorted out.

**Comment by rbharath** [comment]
As a quick update, it appears that the cmake no longer builds tensorflow head due to recent changes in tensorflow :-(. Fixing this looks tricky, and I probably won't have time till next week or so. I'll update this thread then.

**Comment by martinwicke** [comment]
Yeah, we are constantly messing with the location of header files etc.
Sorry. It should stabilize shortly.

**Comment by jendap** [comment]
I forget that cmake build. I will try to get it done by tomorrrow.

**Comment by jendap** [comment]
There is very simple https://github.com/tensorflow/tensorflow/pull/1488. That should be enough for linux cmake cpu build.

I'm have set it up to run on pull request but only when pull request touch cmake. (Bash condition `[ -z "$(git log $master_sha1..HEAD tensorflow/contrib/cmake)" ]` is used for triggering).

That should allow to run cmake tests while not showing errors on non-cmake tests.

Once we get cmake working well we can enable it for all the builds. Then we would love to create windows build as well.

**Comment by jendap** [comment]
@rbharath can you please modify this PR to contain the intended changes only?

**Comment by rbharath** [comment]
@jendap This PR was broken by changes in tensorflow, and fixing the issue looks tricky. I don't have time to work on this for a bit (probably a couple weeks), so I'll close this PR for the time being. I'll open a new PR with fixes if somebody else doesn't fix the issue in the meanwhile.

**Comment by jendap** [comment]
Ok, thanks! I hope somebody will help.

**Pull request closed without merging**